functions/uploader/entities/ia-request.js

```diff
@@ -0,0 +1,97 @@
+const _ = require(`lodash`);
+const debug = require(`debug`)(`ia:uploader:entities:debug`);
+const error = require(`debug`)(`ia:uploader:entities:error`);
+const fetch = require(`node-fetch`);
+const mustache = require('mustache');
+
+const config = require('../../src/config');
+const {postEntitiesToDF, fetchEntitiesFromDF} = require('./entities');
+
+/**
+ * get unique & filtered entities for DialogFlow
+ *
+ * @param docs {array}
+ * @returns entities {array}
+ */
+function getUniqueCreatorsFromIA (docs) {
+  var creators = [];
+  var strCreator = ``;
+  for (let i = 0; i < docs.length; i++) {
+    var creator = docs[i].creator;
+    if (_.isArray(creator)) {
+      for (let i = 0; i < creator.length; i++) {
+        strCreator = creator[i];
+        if (strCreator) {
+          strCreator = strCreator.replace(/[()]+/g, ``);
+          creators.push(strCreator);
+        }
+      }
+    } else {
+      strCreator = creator;
+      if (strCreator) {
+        strCreator = strCreator.replace(/[()]+/g, ``);
+        creators.push(strCreator);
+      }
+    }
+  }
+  return _.uniq(creators);
+}
+
+/**
+ * get entities from InternetArchive
```

Comment:
It looks like it fetches only `creator` field. it should be an option of this function

Reply:
then what should be the key for genres, actually I was considering creator to be the key for genres as well.